ts/setCompletedTodosDeletionListener.ts

```diff
@@ -0,0 +1,34 @@
+import { tasks, todos } from "./todoapp.js";
+import { reassignTodos } from "./reassignTodos.js";
+import { storage } from "./storage.js";
+import { isTypeTodo } from "./setSingleTodoDeletionListener.js";
+
+export const deleteCompletedButton = document.querySelector(
+  ".delete-completed"
+) as HTMLElement;
+
+export function setCompletedTodosDeletionListener() {
+  deleteCompletedButton.addEventListener("click", () => {
+    const allTodos = reassignTodos()[0];
+    allTodos.forEach((todo) => {
+      const currentTodoId = Number(todo.id);
+
+      if (isTypeTodo(tasks)) {
```

Comment:
You are repeating this logic again and again. But in general 1 thing to understand here: you are lying to yourself. 

You `tasks` variable should be always `todos`, because nothing else lives in our local storage under `tasks` key except todos array. 

But you are checking it again and again :)